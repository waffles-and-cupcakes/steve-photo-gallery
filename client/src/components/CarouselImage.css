.main-image {
  height: 70vh;
  cursor: pointer;
}
