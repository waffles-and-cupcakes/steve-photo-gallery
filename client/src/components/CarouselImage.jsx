import React from 'react';
import PropTypes from 'prop-types';
import Slider from 'react-slick';
import style from './CarouselImage.css';

const CarouselImage = (props) => (
  <Slider
    accessability={true}
    swipe={false}
    fade={true}
    lazyLoad={'progressive'}
    asNavFor={props.thumbnail}
    afterChange={(current) => props.setCurrentImageIndex(current)}
    ref={props.setMain} >
    {props.photos.map((photo, index) => (
      <div key={photo.photoUrl + index}>
        <img className={style.mainImage} onClick={() => props.sliderMain.slickNext()} src={`${photo.photoUrl}`} />
      </div>
    ))}
  </Slider>
);

CarouselImage.propTypes = {
  photos: PropTypes.array,
  setMain: PropTypes.func,
  setCurrentImageIndex: PropTypes.func,
  thumbnail: PropTypes.object,
};

export default CarouselImage;
