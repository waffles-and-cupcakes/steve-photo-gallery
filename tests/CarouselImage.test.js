import React from 'react';
import { shallow, mount, render, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import CarouselImage from './CarouselImage.jsx';

configure({adapter: new Adapter() });

const photos = [
  {
    title: 'A photo',
    isVerified: false,
    photoUrl: 'www.example.com/photo',
    thumbnailUrl: 'www.example.com/thumbnail',
    listingId: 1,
  },
  {
    title: 'Another photo',
    isVerified: true,
    photoUrl: 'www.example.com/photo',
    thumbnailUrl: 'www.example.com/thumbnail',
    listingId: 1,
  }
];

test('CarouselImage should render correctly', () => {
  const output = shallow(<CarouselImage photos={photos} />);
  expect(shallowToJson(output)).toMatchSnapshot();
});