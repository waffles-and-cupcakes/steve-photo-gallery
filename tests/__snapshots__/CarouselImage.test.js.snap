// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselImage should render correctly 1`] = `
<Slider
  accessability={true}
  afterChange={[Function]}
  fade={true}
  lazyLoad="progressive"
  swipe={false}
>
  <div
    key="www.example.com/photo0"
  >
    <img
      onClick={[Function]}
      src="www.example.com/photo"
    />
  </div>
  <div
    key="www.example.com/photo1"
  >
    <img
      onClick={[Function]}
      src="www.example.com/photo"
    />
  </div>
</Slider>
`;
